tangelo/toolboxes/ansatz_generator/_qubit_ilc.py

```diff
@@ -127,62 +129,86 @@ def gauss_elim_over_gf2(a_mat, b_vec=None):
             initial solution of A * z. Default, np.zeros((n, 1)).
 
     Returns:
-        numpy array of float: solution for the z vector of dimension (n, )
+        numpy array of int: solution for the z vector of dimension (n, )
     """
 
+    # check that b_vec was provided properly; otherwise initialize as a vector of zeros
     n_rows, n_cols = np.shape(a_mat)
-    z_vals, z_sln, piv_idx = [], [-1] * n_cols, 0
-    # check that b_vec was properly supplied; ortherwise initialize as a vector of zeros
     if not isinstance(b_vec, np.ndarray):
         b_vec = np.zeros((n_rows, 1))
+
+    # append the initial solution vector as the last column of a_mat; update n_cols
     a_mat = np.append(a_mat, b_vec, axis=1)
     n_cols += 1
+
+    # force all entries of a_mat to be either 0 or 1.
+    warnings.warn("Reducing input matrix elements modulo 2 to create a binary matrix.", RuntimeWarning)
```

Comment:
So the default behaviour is that, everytime this function is called, we print a warning on the user screen?

Reply:
Yes -- originally I just changed a_mat to a binary matrix without warning the user. I added this warning message based on how I interpreted one of your PR review comments. Would you prefer to do it another way such as adding a verbose flag?

Reply:
Changed it so that it only prints a message if a_mat is not passed as a binary matrix.